File: TestHost/PerDocumentSequencePoints.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Serialization;
using R4nd0mApps.TddStud10.Common.Domain;

namespace R4nd0mApps.TddStud10.TestHost
{
    [Serializable]
    public class PerDocumentSequencePoints : SerializableDictionary<FilePath, List<SequencePoint>>
    {
        public static PerDocumentSequencePoints Deserialize(string file)
        {
            return Deserialize<PerDocumentSequencePoints>(file);
        }
    }
}

File: Common/Domain/PerDocumentSequencePoints.fs
namespace R4nd0mApps.TddStud10.Common.Domain

open System.Collections.Concurrent
open System

[<Serializable>]
type PerDocumentSequencePoints = 
    inherit DataStoreEntityBase<FilePath, ConcurrentBag<SequencePoint>>
    
    new() = 
        { inherit DataStoreEntityBase<_, _>() }
        then ()
    
    member public t.Serialize path = DataStoreEntityExtensions.Serialize<PerDocumentSequencePoints> path t

    static member public Deserialize path = DataStoreEntityExtensions.Deserialize<PerDocumentSequencePoints> (path)
